lib/TaskProcessing/SummaryProvider.php

```diff
@@ -114,21 +119,40 @@ public function process(?string $userId, array $input, callable $reportProgress)
 			$model = $this->appConfig->getValueString(Application::APP_ID, 'default_completion_model_id', Application::DEFAULT_MODEL_ID) ?: Application::DEFAULT_MODEL_ID;
 		}
 
-		try {
-			if ($this->openAiAPIService->isUsingOpenAi() || $this->openAiSettingsService->getChatEndpointEnabled()) {
-				$completion = $this->openAiAPIService->createChatCompletion($userId, $model, $prompt, null, null, 1, $maxTokens);
-			} else {
-				$completion = $this->openAiAPIService->createCompletion($userId, $prompt, 1, $model, $maxTokens);
+		while (strlen($prompt) > $maxTokens) {
```

Comment:
Could this turn into an infinite loop?

Reply:
If max_tokens is set small enough, I can see it being a possibility. Maybe whenever the output summary is longer than the input prompt, that could be another terminating condition?

Reply:
Intuitively, I'd say the stop condition should be "only one summary is left"

Reply:
But if each chunk has their own summary, what does it mean to have "only one left/remaining"?

Reply:
> But if each chunk has their own summary, what does it mean to have "only one left/remaining"?

Since we expect the summaries to shorten the input, with each iteration there should be less chunks. So, I guess to have only one summary left, there should be only one chunk left.

Reply:
Quick fix is done.

Reply:
Thanks! Works nicely.

Reply:
> Since we expect the summaries to shorten the input

We don't know that for sure. We could add a stop condition: when the number of chunks does not change between 2 iterations.
Wdyt?

Reply:
I still see two issues here:

1.`$maxTokens` counts in tokens, while strlen counts in bytes
2. `$maxTokens` is the maximum amount of output tokens that we ask the model to produce. Why are we using it here to compare it to the length of the input?

Reply:
In the loop, whatever summary output we get becomes the new prompt (input) in the next iteration. Regardless of whether chunking is disabled or not, this is the most reliable terminating condition we have.